### PR TITLE
* add scp command

### PR DESCRIPTION
**Pull request by hjanuschka**
adds an action to scp files from and to remote-hosts

use-case: i am trying to port some of my deployment scripts, and there i needed to fetch & pull files from remote.

samples:
## Uploads-String to a file

``` ruby
scp(
        host: "dev.januschka.com",
        username: "root",
        upload: {
            :src => StringIO.new("SRC"),
            :dst => "/tmp/1.txt"
        }
    )
```
## upload a file

``` ruby
scp(
        host: "dev.januschka.com",
        username: "root",
        upload: {
            :src => "/tmp/1.txt",
            :dst => "/tmp/1.txt"
        }
    )   
```
## download a file

``` ruby
scp(
        host: "dev.januschka.com",
        username: "root",
        download: {
            :src => "/tmp/1.txt",
            :dst => "/tmp/2.txt"
        }
    )   
```
## download a file but not to disk just to a var

``` ruby
file_out = scp(
        host: "dev.januschka.com",
        username: "root",
        download: {
            :src => "/tmp/1.txt"            
        }
    )   
puts "#{file_out.inspect}"  
```
## download a folder

``` ruby
scp(
        host: "dev.januschka.com",
        username: "root",
        download: {
          :src => "/root/dir1",
            :dst => "/tmp/new_dir"      
        }
)
```
## upload a folder

``` ruby
scp(
        host: "dev.januschka.com",
        username: "root",
        upload: {
          :src => "/root/dir1",
            :dst => "/tmp/new_dir"      
        }
)
```

### CODE REVIEW

File: fastlane.gemspec [hunk 1]
```diff
@@ -31,6 +31,8 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'addressable', '~> 2.3.8' # Support for URI templates
   spec.add_dependency 'xcode-install', '~> 1.0.1' # Download new Xcode versions
 
+  spec.add_dependency 'net/scp', '~> 1.2.1' # add SSH gem
```
Comment (KrauseFx) [inline fastlane.gemspec]: Please don't introduce new dependencies. Instead use the `Actions.verify_gem!('cocoapods')` method in your integration. Check out `cocoapods.rb` on how to do that :+1:

**Comment by KrauseFx** [comment]
That's great :rocket: Similar to the other PRs, please remove the comments, add information to the `Actions.md`. The `Actions.md` should be a very basic, very common example.

**Comment by hjanuschka** [comment]
* rubcop is clean
 * tests pass
 * added `Action.md` entry
 * removed new dependencie - checking now using Action.verify_gem


@KrauseFx

File: docs/Actions.md [hunk 2]
```diff
@@ -1746,6 +1746,37 @@ It can process unit test results if formatted as junit report as shown in [xctes
 
 ## Misc
 
+### scp
+
+Allows SCP file transfer.
+
+
+ * Upload file/folder:
```
Comment (KrauseFx) [inline docs/Actions.md]: This is actually a list in markdown a looks weird here :wink:

File: docs/Actions.md [hunk 3]
```diff
@@ -1746,6 +1746,37 @@ It can process unit test results if formatted as junit report as shown in [xctes
...
+
+```ruby
+scp(
+  host: "dev.januschka.com",
+  username: "root",
+  upload: {
+    :src => "/root/dir1",
```
Comment (KrauseFx) [inline docs/Actions.md]: Please change to `src: "value"` (new Ruby syntax)

File: docs/Actions.md [hunk 4]
```diff
@@ -1746,6 +1746,37 @@ It can process unit test results if formatted as junit report as shown in [xctes
...
+```
+
+ * Download file/folder:
+
+```ruby
+scp(
+  host: "dev.januschka.com",
+  username: "root",
+  download: {
+    :src => "/root/dir1",
```
Comment (KrauseFx) [inline docs/Actions.md]: Same here

File: docs/Actions.md [hunk 5]
```diff
@@ -1746,6 +1746,37 @@ It can process unit test results if formatted as junit report as shown in [xctes
...
+  host: "dev.januschka.com",
+  username: "root",
+  upload: {
+    :src => "/root/dir1",
+    :dst => "/tmp/new_dir"      
+  }
+)
+```
+
+ * Download file/folder:
```
Comment (KrauseFx) [inline docs/Actions.md]: You can use `**` before and after this item to make it bold

File: lib/fastlane/actions/scp.rb [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Transfer files via SCP"
+      end
+
+      def self.details
+        "Allows SCP file transfers"
```
Comment (KrauseFx) [inline lib/fastlane/actions/scp.rb]: This doesn't add any extra information. Please either provide more information about how to use it, or remove this string

File: lib/fastlane/actions/scp.rb [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                                       optional: true,
+                                       default_value: true,
+                                       is_string: false
+                                      )
+
+        ]
+      end
+
+      def self.authors
+        # So no one will ever forget your contribution to fastlane :) You are awesome btw!
```
Comment (KrauseFx) [inline lib/fastlane/actions/scp.rb]: Please remove this comment

**Comment by hjanuschka** [comment]
* feedback adopted
 * merge conflict solved

**Comment by KrauseFx** [comment]
Looks good, thanks for addressing my feedback, however we have a merge conflict again :cry:

**Comment by hjanuschka** [comment]
* ok i removed the Actions.md from this branch - as if you merge this one - the other PR's conflict too
 * you can merge this branch, and later on take the right block from   (https://gist.github.com/hjanuschka/96cb3129fd0d839b7a56) - or take a look at my combined docs PR.

https://github.com/fastlane/fastlane/pull/1207

@KrauseFx

**Comment by KrauseFx** [comment]
Perfect, thank you! :+1:

**Pull request merged**